modules/http4s/src/main/scala/com.snowplowanalytics.iglu/client/resolver/registries/Http4sRegistryLookup.scala

```diff
@@ -12,31 +12,30 @@
  */
 package com.snowplowanalytics.iglu.client.resolver.registries
 
-import scala.util.control.NonFatal
-
-import cats.implicits._
 import cats.data.EitherT
-import cats.effect.Sync
-
+import cats.effect.Concurrent
+import cats.effect.kernel.Async
+import cats.implicits._
+import com.snowplowanalytics.iglu.core.circe.CirceIgluCodecs._
+import com.snowplowanalytics.iglu.core.{SchemaKey, SchemaList}
 import io.circe.Json
-
-import org.http4s.{EntityDecoder, Header, Headers, Method, Request, Status, Uri}
 import org.http4s.circe._
 import org.http4s.client.{Client => HttpClient}
+import org.http4s.{EntityDecoder, Header, Headers, Method, Request, Status, Uri}
+import org.typelevel.ci.CIString
 
-import com.snowplowanalytics.iglu.core.{SchemaKey, SchemaList}
-import com.snowplowanalytics.iglu.core.circe.CirceIgluCodecs._
+import scala.util.control.NonFatal
 
 object Http4sRegistryLookup {
 
-  def apply[F[_]: Sync](client: HttpClient[F]): RegistryLookup[F] =
+  def apply[F[_]: Async](client: HttpClient[F]): RegistryLookup[F] =
```

Comment:
Have you considered changing this to `F[_]: Concurrent: fs2.io.file.Files` ?

Not sure yet how big the win would be. If it means we can instantiate `Client` without `Sync`/`Async` at all - that's great and I'd stick with it. If we still need to use impure type classes for cache - it's probably not worth it (especially given that we need to implement `embeddedLookup` duplicating the original function)

Reply:
Is it possible to read resources (for embedded repo) in RT manner relying only on `Files` and without `Sync`? Looking at the `Files` API you need to pass `Path` to given file and in order to do that you need to call sth like: `getClass.getResource(pathToFile)` which I believe breaks RT.

And as you mentioned we would still need to support:

* Impure implementation of `embeddedLookup` for `Id` RegistryLookup instance.
* Pure implementation of `embeddedLookup` based on `Sync` for RT RegistryLookup instance from core module. We could also replace it with the new one based on `Files`, but `Sync` would still be required for http lookup (because of scalaj-http). We would end up with sth like ```implicit def ioLookupInstance[F[_]: Sync: Files]: RegistryLookup[F]```. BTW maybe we should deprecate this instance? 

So I think it's not worth it.

Also, instead of `Async` we can use `F[_]: Concurrent: Sync` :)

Reply:
That all make sense!